### ISSUE TITLE
Desktop Icons Disappear on Multi-Monitor

### ISSUE THREAD

**Issue by tiredofit** [linuxmint/Cinnamon#5978]
Cinnamon 3.2
OS: Arch
X; Nvidia Driver

Running a triple monitor setup, with icons on all 3 screens. Randomly icons disappear.
In order to restore, required to visit Desktop Settings, Disable Icons on all Monitors, and then Reenable Icons on all monitors.

**Comment by Wini-Buh** [comment]
I think it's the same problem I described here #5973. I could reproduce it easy with help of a virtual machine (I used VirtualBox to check if the problem is reproducible).

At least your work-around to disable icons on all monitors, and then re-enable icons on all monitors works for me (at the moment).

**Comment by itzexor** [comment]
Pressing F5 with the desktop focused will refresh it and make the icons show up. Not a solution but a more convenient workaround for the time being.

**Comment by moretocome** [comment]
Same problem as described, thou disabling and re-enabling the icons does nothing, nor does pressing F5 with desktop focused.
Cinnamon 3.07
Linux Mint 18
Nvidia driver

**Comment by xonxor** [comment]
I can confirm this problem with my 3 monitor setup as well. Both above workarounds provided by OP and @itexor work for me (certainly not long term solutions). This issue first appeared when I upgraded to 18.1 using the upgrade manager. Never had any problems of this sort in Linux Mint 18. The Cinnamon version I'm using is 3.2.7 and I use the Nvidia proprietary driver.

**Comment by carontiello** [comment]
Same problem and pressing F5 on desktop focused, in the meantime, work for me.
Cinnamon 3.2.7
Linux Mint 18.1 x64
Intel Integrated Graphics Controller with triple monitor

**Comment by jameshibbard** [comment]
Same problem here. Mint 18.1, cinnamon. F5 fixes it for me, but more often than not the icons are in random positions.

**Comment by laurent-steamlead** [comment]
same problem, 3 monitors, icons disappear randomly, F5 or changing icon settings can fix it

**Comment by primawill** [comment]
I am also having this problem. Linux Mint 18.1 Dual monitors, AMD Phenom II x4,Nvidia GTX260, Cinnamon 3.2.7

**Comment by jameshibbard** [comment]
Bizarrely, I have found that arranging desktop icons in vertical lines (not horizontal) means that they don't get rearranged when I press F5

**Comment by mariuszpass** [comment]
I am also having this problem. Dual monitors with diffrent resolution:
first one: 1920x1080, second one 1080x1920

**Comment by primawill** [comment]
As it turns out, I think I may have found out what is causing the problem.  I prefer to have my primary monitor on the Right side, and use the left side as my secondary. Upon switching the left to primary and right to secondary the issue went away. 

So I experimented with swapping the cables on the monitors thinking whichever monitor was plugged into the DVI-1 spot should be as the primary. That is not the case. Problem still persisted whenever the right side monitor was made the primary, even after swapping the cables.

So my conclusion: If you are using dual monitors, the left side should be the primary and the right side secondary.

**Comment by nsoulantikas** [comment]
My setup:
Kernel: 4.4.0-77-generic x86_64 (64 bit gcc: 5.4.0)
Desktop: Cinnamon 3.2.7 (Gtk 3.18.9-1ubuntu3.3)
Distro: Linux Mint 18.1 Serena
Graphics:  Card: Advanced Micro Devices [AMD/ATI] Thames [Radeon HD 7500M/7600M Series] bus-ID: 01:00.0
Display Server: X.Org 1.18.4 drivers: ati,radeon (unloaded: fbdev,vesa)
Resolution: 1366x768@60.02hz, 1920x1080@60.00hz, 1920x1080@60.00

3 monitors and the desktop icons disappear when i turn off the primary monitor. It mostly happens for the right click not to work. Initially i switch the monitor settings (where to show icons) but the right click stops functioning.

**Comment by feugy** [comment]
What Nsoulantikas describe is still with cinammon 18.3

**Comment by t-hofmann** [comment]
Setup:
- two monitors
- LinuxMint 18.3, Cinnamon

Symptoms on right monitor:
- no right click
- no desktop icons
regardless of which monitor is the primary.

Using xrandr to scale and position, but no difference if beeing executed or not.

**Comment by FrancescoBorzi** [comment]
I have the same issue when using Nemo on Ubuntu 20.04

**Comment by FrancescoBorzi** [comment]
I'm having the same issue using Nemo on Ubuntu 20.04 with 2 displays.

I managed to mirror the icons by right-clicking on the empty desktop and choosing "Customize". However, now I can no longer find a way to open the "Customize" options of my desktop.

**Comment by aplatypus** [comment]
Yes -- #metoo since the recent updates this week.  Desktop icons vanish and application windows all "gather in the same monitor" with a multim monitor set-up.

ANYWAY ,,, I would like to float the concept that Desktop configuration SHALL BE transparent.  

My second point is more useful.  You can "get your icons/shorcuts" back as follows.

- identify the file matching the missing desktop icon
- `ls -l ~/Desktop/*.desktop`
- Rename it ...
   - `mv -v  ~/Desktop/01_example.desktop  ~/Desktop/02_example.desktop`
- '_Something_' picks that file name change and the logo reappears!!
  - Unfortunately `touch` does not work.  It must be a rename (`mv`) or a copy (`cp`).
  - Go figure!!

I'm also a little mystified by the "community attitude" to Desktop as a concept.  Maybe I need a blog.  My second job was at a university and they had two(2) Lisa workstations.  They had a desktop.  After working on a Cray design CD mainframe it was like a shot of some kind of illegal substance!

I have never been clear on any direction that chooses to 'remove' a desktip as a metaphor without offering an alternative.  In later life I studied the area of behaviour about consumer choices.  So far no one ever created a ground swell of adoption with designs that refused to meet the preferences of the 'influencers' and 'mainstream' customer.

Remember the take-up and resounding success of Windows 8!???
-  NO?

Exactly the point.

**Comment by robertwhitton** [comment]
I struggle to believe that this is still broken since it was first reported in 2016 - this is pretty basic stuff.

Desktop icons are basically an unusable feature at the moment as far as I can see. I have all of the issues mentioned above. With icons disappearing, icons randomly moving about etc. The F5 refresh will sometimes recover icons but the original ordering is lost.

There are also other problems. The icon ordering changes whether they are on the primary monitor or not. If I ask for the icons to be on both monitors then I get some of them on the primary and some of them on the non-primary but none of the icons appear on both. I can't see any commonality between the icons it puts on each monitor - it's truly bizarre.

I'm more than willing to help debug this issue if there are any tests that I could run or any command output that would prove useful.

It's pretty easy to provoke problems by using the desktop settings to move the icons between monitors.

I have attached some screenshots showing some of the problems using this technique. These are in time order. I start with my desktop icons ordered how I like with them on the non-primary monitor (left half of the screen capture). Then I move the icons to the primary monitor. Oops they all get reordered. Now I move them back to the non-primary - they get even crazier. Finally I ask for the icons to be on both monitors and the set of icons seems to get randomly split across both monitors.

![Screenshot from 2020-12-16 19-26-58](https://user-images.githubusercontent.com/19554382/102396821-eefb5f00-3fd4-11eb-8727-50f137cd10c8.png)
![Screenshot from 2020-12-16 19-27-29](https://user-images.githubusercontent.com/19554382/102396849-f7539a00-3fd4-11eb-8bb8-f70d6a897b7b.png)
![Screenshot from 2020-12-16 19-27-59](https://user-images.githubusercontent.com/19554382/102396859-fa4e8a80-3fd4-11eb-8937-a014bae21133.png)
![Screenshot from 2020-12-16 19-28-20](https://user-images.githubusercontent.com/19554382/102396885-03d7f280-3fd5-11eb-8902-cc6afd7cb9de.png)

**Comment by robertwhitton** [comment]
I should have added that I obtained the above results in LM20/Cinnamon 4.6.7
